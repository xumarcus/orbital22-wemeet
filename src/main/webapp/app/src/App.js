import * as React from 'react';
import CssBaseline from '@mui/material/CssBaseline';
import NavigationBar from "./components/NavigationBar";
import Typography from "@mui/material/Typography";
import { Grid } from "@mui/material";
import UserProfileExample from "./components/UserProfileExample";

const App = () => {
    return (
        <React.Fragment>
            <CssBaseline />
            <NavigationBar />
            <Grid
                container
                direction={"column"}
                alignItems={"center"}
                justifyContent={"center"}
                style={{ minHeight: '80vh' }}>
                <Grid item>
                    <Typography variant="h1" component="div">
                        <UserProfileExample />
                    </Typography>
                </Grid>
            </Grid>
        </React.Fragment>
    );
};

export default App;
