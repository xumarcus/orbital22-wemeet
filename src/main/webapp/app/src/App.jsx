import * as React from 'react'
import { useEffect, useState } from 'react'
import CssBaseline from '@mui/material/CssBaseline'
import { BrowserRouter, Route, Routes } from 'react-router-dom'
import Footer from './components/footer/Footer'
import NavBar from './components/navbar/NavBar'
import Container from '@mui/material/Container'
import AppContext, { defaultAppContext } from './core/AppContext'
import { ErrorBoundary } from 'react-error-boundary'
import ErrorFallback from './components/core/ErrorFallback'

import './App.css'

import Home from './pages/Home'
import Dashboard from './pages/Dashboard'
import MeetingEdit from './pages/MeetingEdit'
import { API } from './core/const'
import ajax from './core/ajax'
import MeetingRank from './pages/MeetingRank'
import MeetingViewSolution from './pages/MeetingViewSolution'

const App = () => {
  const [context, setContext] = useState(defaultAppContext)

  useEffect(() => {
    ajax('GET')(API.ME).then(user => setContext(c => ({ ...c, user })))
  }, [])

  return (
    <>
      <CssBaseline />

      <BrowserRouter>
        <AppContext.Provider value={{ context, setContext }}>
          <NavBar />
          <Container maxWidth='xl'>
            <ErrorBoundary FallbackComponent={ErrorFallback}>
              <Routes>
                <Route path='' element={<Home />} />
                <Route path='dashboard' element={<Dashboard />} />
                <Route path='meeting'>
                  <Route path=':meetingId/edit' element={<MeetingEdit />} />
                  <Route path=':meetingId/rank' element={<MeetingRank />} />
                  <Route
                    path=':meetingId/view-solution'
                    element={<MeetingViewSolution />}
                  />
                </Route>
                <Route path='*' element={<ErrorFallback />} />
              </Routes>
            </ErrorBoundary>
          </Container>
        </AppContext.Provider>
      </BrowserRouter>

      <Footer />
    </>
  )
}

export default App
