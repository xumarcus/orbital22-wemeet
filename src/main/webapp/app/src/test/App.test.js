import React from 'react'
import { render } from '@testing-library/react'
import App from '../App'

/*
  There is not much point of unit testing when there is not much
  custom UI logic or client side validation. End-to-end tests are
  far more useful, but (...) asks us to have these tests anyway...
 */

describe('Main page', () => {
  it('should render correctly', () => {
    const { asFragment, getByText } = render(<App />)
    expect(asFragment()).toMatchSnapshot()
    expect(getByText('Easy')).toBeInTheDocument()
  })
})
