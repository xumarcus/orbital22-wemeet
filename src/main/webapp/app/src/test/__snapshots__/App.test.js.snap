// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page should render correctly 1`] = `
<DocumentFragment>
  <header
    class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation4 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionStatic css-uv47wz-MuiPaper-root-MuiAppBar-root"
  >
    <div
      class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
    >
      <div
        class="MuiToolbar-root MuiToolbar-regular css-r6ewbb-MuiToolbar-root"
      >
        <a
          href="/"
        >
          <img
            alt="WeMeet"
            class="MuiBox-root css-1pgf9wo"
            src="logo_banner.png"
          />
        </a>
        <div
          class="MuiBox-root css-1lvtzne"
        >
          <button
            aria-controls="menu-appbar"
            aria-haspopup="true"
            aria-label="account of current user"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorSecondary MuiIconButton-sizeLarge css-1f88mc5-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="MenuIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-wfg4so"
        >
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-17e5olp-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            features
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-17e5olp-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            guide
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
          <button
            class="MuiButton-root MuiButton-text MuiButton-textPrimary MuiButton-sizeMedium MuiButton-textSizeMedium MuiButtonBase-root  css-17e5olp-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            about
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-2uchni"
        >
          <button
            aria-label="Open settings"
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-av538e-MuiButtonBase-root-MuiIconButton-root"
            data-mui-internal-clone-element="true"
            tabindex="0"
            type="button"
          >
            <div
              class="MuiAvatar-root MuiAvatar-circular css-1wlk0hk-MuiAvatar-root"
            >
              <img
                alt="Anonymous"
                class="MuiAvatar-img css-1pqm26d-MuiAvatar-img"
                src="/avatar-does-not-exist"
              />
            </div>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <div
          class="MuiBox-root css-n085mf"
        >
          <button
            class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeLarge MuiButton-containedSizeLarge MuiButtonBase-root  css-mkazzp-MuiButtonBase-root-MuiButton-root"
            tabindex="0"
            type="button"
          >
            Start Now
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
      </div>
    </div>
  </header>
  <div
    class="MuiContainer-root MuiContainer-maxWidthXl css-19r6kue-MuiContainer-root"
  >
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-5 css-1ld3b9g-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 css-1k7sk4d-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-z1460q"
        >
          <h3
            class="MuiTypography-root MuiTypography-h3 MuiTypography-alignCenter css-47suzj-MuiTypography-root"
          >
            Scheduling 
            <br />
            Made 
            <strong>
              Easy
            </strong>
          </h3>
        </div>
        <div
          class="MuiBox-root css-1qapnz0"
        >
          <span
            class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-6 css-1k7sk4d-MuiGrid-root"
      >
        <img
          class="MuiBox-root css-8atqhb"
          src="cover_img.png"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root"
        >
          Item 1?
        </h5>
        <div
          class="MuiBox-root css-1qapnz0"
        >
          <span
            class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root"
        >
          Item 2?
        </h5>
        <div
          class="MuiBox-root css-1qapnz0"
        >
          <span
            class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </span>
        </div>
      </div>
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-md-4 css-1ube6tp-MuiGrid-root"
      >
        <h5
          class="MuiTypography-root MuiTypography-h5 MuiTypography-alignCenter css-ph2xsj-MuiTypography-root"
        >
          Item 3?
        </h5>
        <div
          class="MuiBox-root css-1qapnz0"
        >
          <span
            class="MuiTypography-root MuiTypography-body css-37jdci-MuiTypography-root"
          >
            Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua. Ut enim ad minim veniam, quis nostrud exercitation ullamco laboris nisi ut aliquip ex ea commodo consequat.
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-regular css-iz63q-MuiToolbar-root"
  >
    <p
      class="MuiTypography-root MuiTypography-body1 MuiTypography-alignCenter css-1uwgr7b-MuiTypography-root"
    >
      An Orbital 2022 Project
    </p>
  </div>
</DocumentFragment>
`;
